shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -634,13 +657,51 @@ public void didChangeEditingState(
                 && composingEnd == mLastKnownFrameworkTextEditingState.composingEnd);
     if (!skipFrameworkUpdate) {
       Log.v(TAG, "send EditingState to flutter: " + mEditable.toString());
-      textInputChannel.updateEditingState(
-          inputTarget.id,
-          mEditable.toString(),
-          selectionStart,
-          selectionEnd,
-          composingStart,
-          composingEnd);
+
+      if (configuration.enableDeltaModel) {
+        // Make sure last delta has the most up to date composing region.
+        // This can happen when the IME is restarted because the framework has changed the
+        // composing region. We do not have access to the new composing region until after
+        // the restart has completed.
```

Comment:
Tricky.

Reply:
For some reason when I capture it in `ListenableEditingState` the composing region is still `(-1,-1)`. It is not until I arrive at `ListenableEditingState` that I'm able to grab the most up to date composing regions. I think this is somewhat consistent to how `updateEditingState` works, where we send the composing/selection regions we grab in `didChangeEditingState`.

I have observed that this is only the case when the composing region restarts. In other cases the selection/composing regions are up to date after the editable has been mutated with `replace`.